src/main/java/im/toduck/domain/schedule/domain/usecase/ScheduleUseCase.java

```diff
@@ -43,4 +50,45 @@ public ScheduleInfoResponse getSchedule(Long userId, Long scheduleRecordId) {
 			.orElseThrow(() -> CommonException.from(ExceptionCode.NOT_FOUND_USER));
 		return scheduleService.getSchedule(scheduleRecordId);
 	}
+
+	@Transactional
+	public void completeSchedule(Long userId, ScheduleCompleteRequest scheduleCompleteRequest) {
+		userService.getUserById(userId)
+			.orElseThrow(() -> CommonException.from(ExceptionCode.NOT_FOUND_USER));
+		Schedule schedule = scheduleService.getScheduleById(scheduleCompleteRequest.scheduleId())
+			.orElseThrow(() -> CommonException.from(ExceptionCode.NOT_FOUND_SCHEDULE));
+		scheduleRecordService.getScheduleRecordWithSchedule(userId, scheduleCompleteRequest)
+			.ifPresentOrElse((scheduleRecord) -> {
+				scheduleRecordService.completeScheduleRecord(scheduleRecord, scheduleCompleteRequest);
+			}, () -> {
+				scheduleRecordService.createScheduleRecord(schedule, scheduleCompleteRequest);
+			});
+	}
+
+	@Transactional
+	public void deleteSchedule(Long userId, ScheduleDeleteRequest scheduleDeleteRequest) {
+		userService.getUserById(userId)
+			.orElseThrow(() -> CommonException.from(ExceptionCode.NOT_FOUND_USER));
+		Schedule schedule = scheduleService.getScheduleById(scheduleDeleteRequest.scheduleId())
+			.orElseThrow(() -> CommonException.from(ExceptionCode.NOT_FOUND_SCHEDULE));
+
+		if (isSingleDaySchedule(schedule)) {
+			scheduleService.deleteSingleDaySchedule(schedule, scheduleDeleteRequest);
+			log.info("반복 X 하루 일정 삭제 성공 : {}", scheduleDeleteRequest.scheduleId());
+		} else {
+			if (scheduleDeleteRequest.isOneDayDeleted()) {
+				scheduleService.deleteOneDayDeletionForRepeatingSchedule(schedule, scheduleDeleteRequest);
+				log.info("반복 일정 중 하루 삭제 성공 : {}", scheduleDeleteRequest.scheduleId());
+			} else {
+				scheduleService.deleteAfterDeletionForRepeatingSchedule(schedule, scheduleDeleteRequest);
+				log.info("반복 일정 중 기간 삭제 성공 : {}", scheduleDeleteRequest.scheduleId());
+			}
+		}
```

Comment:
```suggestion
		if (isSingleDaySchedule(schedule)) {
			scheduleService.deleteSingleDaySchedule(schedule, scheduleDeleteRequest);
			log.info("반복 X 하루 일정 삭제 성공 : {}", scheduleDeleteRequest.scheduleId());
			return;
		}
	    if (scheduleDeleteRequest.isOneDayDeleted()) {
		    scheduleService.deleteOneDayDeletionForRepeatingSchedule(schedule, scheduleDeleteRequest);
		    log.info("반복 일정 중 하루 삭제 성공 : {}", scheduleDeleteRequest.scheduleId());
		    reteurn;
	    }
		scheduleService.deleteAfterDeletionForRepeatingSchedule(schedule, scheduleDeleteRequest);
		log.info("반복 일정 중 기간 삭제 성공 : {}", scheduleDeleteRequest.scheduleId());
```
p5: 취향 차이일 수 있지만, 개인적으로 Early Return 사용하면 뎁스도 낮아지고 좋을 것 같은데 어떻게 생각하시나요?